astropy/utils/diff.py

```diff
@@ -0,0 +1,155 @@
+import difflib
+import functools
+import operator
+from functools import reduce
+from itertools import islice
+
+import numpy as np
+
+from .misc import indent
+
+__all__ = ['fixed_width_indent', 'diff_values', 'report_diff_values',
+           'where_not_allclose']
+
+# Smaller default shift-width for indent
+fixed_width_indent = functools.partial(indent, width=2)
+
+
+def diff_values(a, b, rtol=0.0, atol=0.0):
+    """
+    Diff two scalar values. If both values are floats, they are compared to
+    within the given absolute and relative tolerance.
+
+    Parameters
+    ----------
+    a, b : int, float, str
+        Scalar values to compare.
+
+    rtol, atol : float
+        Relative and absolute tolerances as accepted by
+        :func:`numpy.allclose`.
+
+    Returns
+    -------
+    is_different : bool
+        `True` if they are different, else `False`.
+
+    """
+    if isinstance(a, float) and isinstance(b, float):
+        if np.isnan(a) and np.isnan(b):
+            return False
+        return not np.allclose(a, b, rtol=rtol, atol=atol)
+    else:
+        return a != b
+
+
+def report_diff_values(fileobj, a, b, ind=0):
+    """
+    Write a diff report between two values to the specified file-like object.
+
+    Parameters
+    ----------
+    fileobj : obj
+        File-like object to write to.
+        To write to terminal, use ``sys.stdout``.
+
+    a, b
+        Values to compare. Anything that can be turned into strings
+        and compared using :py:mod:`difflib` should work.
+
+    ind : int
```

Comment:
Maybe `indent` or `indent_width` to have something more explicit ? (`ind` usually makes me think about indices)

Reply:
Or `pad` (to keep things short 😄 )?